modules/apps/knowledge-base/knowledge-base-web/src/main/java/com/liferay/knowledge/base/web/internal/item/selector/KBArticleItemSelectorView.java

```diff
@@ -112,11 +120,24 @@ public void renderHTML(
 			PortletURL portletURL, String itemSelectedEventName, boolean search)
 		throws IOException, ServletException {
 
-		_itemSelectorViewDescriptorRenderer.renderHTML(
-			servletRequest, servletResponse, infoItemItemSelectorCriterion,
-			portletURL, itemSelectedEventName, search,
-			new KBArticleItemSelectorViewDescriptor(
-				(HttpServletRequest)servletRequest, portletURL));
+		KBArticleItemSelectorViewDisplayContext
+			kbArticleItemSelectorViewDisplayContext =
+				new KBArticleItemSelectorViewDisplayContext(
+					(HttpServletRequest)servletRequest,
+					infoItemItemSelectorCriterion, itemSelectedEventName,
+					portletURL, search);
+
+		servletRequest.setAttribute(
+			KBWebKeys.KB_ARTICLE_ITEM_SELECTOR_VIEW_DISPLAY_CONTEXT,
```

Comment:
Instead of creating a web key for the display context, the current approach is to use the class name as attribute name; you'll see old style web keys used because we haven't updated old code.
```
servletRequest.setAttribute(KBArticleItemSelectorViewDisplayContext.class.getName(), ...);
```

Reply:
👍🏻